integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/kafka/KafkaResource.java

```diff
@@ -0,0 +1,155 @@
+import org.keycloak.representations.AccessTokenResponse;
+
+import com.openshift.cloud.api.kas.auth.AclsApi;
+import com.openshift.cloud.api.kas.auth.TopicsApi;
+import com.openshift.cloud.api.kas.auth.invoker.ApiClient;
+import com.openshift.cloud.api.kas.auth.invoker.ApiException;
+import com.openshift.cloud.api.kas.auth.invoker.Configuration;
+import com.openshift.cloud.api.kas.auth.invoker.auth.OAuth;
+import com.openshift.cloud.api.kas.auth.models.AclBinding;
+import com.openshift.cloud.api.kas.auth.models.AclOperation;
+import com.openshift.cloud.api.kas.auth.models.AclOperationFilter;
+import com.openshift.cloud.api.kas.auth.models.AclPatternType;
+import com.openshift.cloud.api.kas.auth.models.AclPatternTypeFilter;
+import com.openshift.cloud.api.kas.auth.models.AclPermissionType;
+import com.openshift.cloud.api.kas.auth.models.AclPermissionTypeFilter;
+import com.openshift.cloud.api.kas.auth.models.AclResourceType;
+import com.openshift.cloud.api.kas.auth.models.AclResourceTypeFilter;
+import com.openshift.cloud.api.kas.auth.models.NewTopicInput;
+import com.openshift.cloud.api.kas.auth.models.TopicSettings;
+
+import io.restassured.http.ContentType;
+
+import static io.restassured.RestAssured.given;
+
+public class KafkaResource {
+    public static void createKafkaTopic(String topicName) {
+        KafkaConnectionParameters kafkaConnectionParameters = KafkaConnectionParameters.getInstance();
+        ApiClient apiClient = createApiClient();
+        TopicsApi topicsApi = new TopicsApi(apiClient);
+        NewTopicInput newTopicInput = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(1));
+        try {
+            topicsApi.createTopic(newTopicInput);
+        } catch (ApiException e) {
+            throw new RuntimeException("Error creating new topic", e);
+        }
+
+        AclsApi aclsApi = new AclsApi(apiClient);
+        AclBinding aclBindingRead = new AclBinding()
+                .resourceType(AclResourceType.TOPIC)
+                .resourceName(topicName)
+                .patternType(AclPatternType.LITERAL)
+                .principal("User:" + kafkaConnectionParameters.getOpsClientID())
+                .operation(AclOperation.READ)
+                .permission(AclPermissionType.ALLOW);
+        AclBinding aclBindingWrite = new AclBinding()
+                .resourceType(AclResourceType.TOPIC)
+                .resourceName(topicName)
+                .patternType(AclPatternType.LITERAL)
+                .principal("User:" + kafkaConnectionParameters.getOpsClientID())
+                .operation(AclOperation.WRITE)
+                .permission(AclPermissionType.ALLOW);
+        try {
+            aclsApi.createAcl(aclBindingRead);
+            aclsApi.createAcl(aclBindingWrite);
+        } catch (ApiException e) {
+            throw new RuntimeException("Error assigning ACL to topic", e);
+        }
```

Comment:
I would make each call in a different `try/catch` because we should try to revert the part that has been already created (i.e. the topic and the first ACL).
What I mean is a `cleanup` method that tries to delete the other resources before stopping the execution

Reply:
@danielezonca Thank you. Good point. Now all previously allocated resources should be deleted in case of exception.